python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -196,15 +216,19 @@ def fill_or_return_status_for(key: AssetKey):
     return final
 
 
-def get_assets_live_info(graphene_info, step_keys_by_asset: Mapping[AssetKey, List[str]]):
+def get_assets_latest_info(graphene_info, step_keys_by_asset: Mapping[AssetKey, List[str]]):
     from ..schema.asset_graph import GrapheneAssetLatestInfo
     from ..schema.logs.events import GrapheneMaterializationEvent
     from ..schema.pipelines.pipeline import GrapheneRun
 
     instance = graphene_info.context.instance
 
     asset_nodes = get_asset_nodes_by_asset_key(graphene_info)
-    asset_records = instance.get_asset_records(asset_nodes.keys())
+    asset_record_keys_needed = add_all_upstream_keys(asset_nodes, step_keys_by_asset.keys())
+    if not asset_record_keys_needed:
+        return []
+
```

Comment:
could this be the logic in get_asset_records instead / as well? Seems like a bit of a landmine to have passing in the empty list be treated identical as passing in None

Reply:
Ahh yeah, I was afraid there might be implications of changing the behavior there but I'll review the callsites and see. Would definitely be nice to do an audit of places we say `if not array:` in these resolvers